src/main/java/org/broadinstitute/consent/http/service/DacService.java

```diff
@@ -179,9 +183,22 @@ public void updateDac(String name, String description, String email, Integer dac
     dacDAO.updateDac(name, description, email, updateDate, dacId);
   }
 
-  public void deleteDac(Integer dacId) {
-    dacDAO.deleteDacMembers(dacId);
-    dacDAO.deleteDac(dacId);
+  public void deleteDac(Integer dacId) throws IllegalArgumentException, SQLException {
+    Dac fullDac = dacDAO.findById(dacId);
+    List<DataAccessAgreement> daas = daaDAO.findAll();
+    List<Dac> dacs = dacDAO.findAll();
+    Optional<Dac> broadDac = dacs.stream()
+        .filter(dac -> dac.getName().toLowerCase().contains("broad") && dac.getDacId().equals(dacId)).findFirst();
+    if (broadDac.isPresent()) {
+      throw new IllegalArgumentException("This is the Broad DAC, which can not be deleted.");
+    }
+    List<DataAccessAgreement> filteredDaas = daas.stream()
+        .filter(daa -> daa.getInitialDacId().equals(dacId) && ! daaService.isBroadDAA(daa.getDaaId(), daas, dacs)).toList();
```

Comment:
I think there is some duplicated work here. First, we grab all the dacs to find out if we have the Broad one.
Then, again, in the filter, we're checking for a similar thing with `daaService.isBroadDAA`. If we've errored out with a Broad DAC, then it stands to follow that we are only working with non-Broad DAAs.